gitpod-io/gitpod
# [public-api-server] Harmonise configuration structure

## Description
This PR harmonises the structure of the `public-api-server` so that
- changes to its configuration require a change in the installer: before, we relied solely on unit tests on command line flags. Now, in addition, we rely on structural changes, too.
- it looks and feels a lot more like existing components, which helps folks navigate the codebase

## How to test
- ensure the public-api-server still works as intended
- Run ` go run main.go api workspaces get --id 123 --token foo --address api.cw-api-cfg.preview.gitpod-dev.com:443` to validate API is reachable - you will get unathorized due to token being dummy but this is the expected response and the service works.

## Release Notes

```release-note
NONE
```
